File: android/app/src/main/java/com/studyapp/sync/SyncConflictModels.kt
package com.studyapp.sync

enum class SyncConflictField {
    NAME,
    NOTE,
    CURRENT_PAGE,
    TOTAL_PAGES,
    TOTAL_PROBLEMS,
    PROBLEM_RECORDS,
    PROBLEM_CHAPTERS,
    DELETION,
    PROBLEM_REVIEW_STATE,
    OTHER
}

enum class SyncConflictResolutionStrategy {
    KEEP_LOCAL,
    KEEP_REMOTE,
    KEEP_MERGED
}

data class SyncConflict(
    val kind: SyncEntityKind,
    val syncId: String,
    val title: String,
    val summary: String,
    val conflictFields: List<SyncConflictField>,
    val baseJson: String?,
    val localJson: String,
    val remoteJson: String,
    val suggestedMergedJson: String,
    val detectedAt: Long
) {
    val documentId: String get() = "${kind.rawValue}-$syncId"
}

data class SyncConflictResolution(
    val kind: SyncEntityKind,
    val syncId: String,
    val strategy: SyncConflictResolutionStrategy
)

File: ios/StudyApp/Data/Sync/SyncConflictModels.swift
import Foundation

enum SyncConflictField: String, Codable, CaseIterable {
    case name
    case note
    case currentPage
    case totalPages
    case totalProblems
    case problemRecords
    case problemChapters
    case deletion
    case problemReviewState
    case other
}

enum SyncConflictResolutionStrategy: String, Codable, CaseIterable {
    case keepLocal
    case keepRemote
    case keepMerged
}

/// A single unresolved entity-level sync conflict awaiting user choice.
struct SyncConflict: Identifiable, Codable, Equatable {
    var id: String { "\(kind.rawValue)-\(syncId)" }
    var kind: SyncEntityKind
    var syncId: String
    var title: String
    var summary: String
    var conflictFields: [SyncConflictField]
    var baseJson: String?
    var localJson: String
    var remoteJson: String
    var suggestedMergedJson: String
    var detectedAt: Int64

    var documentId: String { "\(kind.rawValue)-\(syncId)" }
}

struct SyncConflictResolution: Codable, Equatable {
    var kind: SyncEntityKind
    var syncId: String
    var strategy: SyncConflictResolutionStrategy
}
